lib/tds.js

```diff
@@ -9,19 +9,12 @@ const {
     generateRequestDomainsByTrackerDomain
 } = require('./utils')
 
-// Custom actions support features like click-to-load, allowing for rules that
-// are defined by the regular TDS and applied by the main blocking ruleset,
-// but also allow for an opposite rule to be applied to nullify / modify that rule.
-// For example, for click-to-load we need to block and redirect certain requests by
-// default, then allow those requests for a given tab if the user wants to reveal
-// something that was blocked
-
-// This map describes the currently supported custom actions and their corresponding
-// DNR action (and inverse)
-const supportedCustomActions = new Map([
-    ['block-ctl-yt', { action: 'block', inverseAction: 'allow' }],
-    ['block-ctl-fb', { action: 'block', inverseAction: 'allow' }]
-])
+// Tracker entry rules with an action starting with this prefix relate to the
+// "Click to Load" feature. The feature blocks third-party embedded content, but
+// provides the user an option to load the content again. To facilitate that,
+// both the blocking/redirecting declarativeNetRequest rules and inverse
+// allowing declarativeNetRequest rules are generated.
+const clickToLoadActionPrefix = 'block-ctl-'
```

Comment:
Well, my thinking is that we should keep this as simple as possible. Right now this is only for Click to Load, so we should be clear about that and worry about handling the general case if/when that becomes necessary.

Reply:
This looks fine to me. We might want to move this data to the blocklist at some point, or was there a reason we didn't add it there?

Reply:
At the point we generate the declarativeNetRequest rules for tracker blocking (and Click to Load), we don't know which Click to Load rule actions will be active. Especially since in the future they will be generated at build time and bundled as static rules.

So the idea here is to generate the rules for all Click to Load actions, using this known prefix, then returning them all. The extension can then decide which blocking/redirecting rules to use, by taking care to enable the inverse rules for the blocking/redirecting rules that shouldn't be active. That will also let us handle things like first-party websites, where one Click to Load entity (and therefore the corresponding rule actions) should never be active.